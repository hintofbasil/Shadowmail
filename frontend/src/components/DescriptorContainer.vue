<template>
  <div class="background">
    <div class="container">
      <div class="row descriptor-container">
        <DescriptorElement
          v-for="element of content"
          :element="element"
          :key="element.title"
        />
      </div>
    </div>
  </div>
</template>

<script>
const CONTENT = [
  {
    title: "Simple",
    symbol: "done",
    description: `
      <span>
        ShadowMail is simple to use.
        Simply enter your email address and start using your
        private, virtual email address immediately.
      </span>
    `
  },
  {
    title: "Secure",
    symbol: "lock",
    description: `
      <span>
        ShadowMails is 100%
        <a href="https://github.com/hintofbasil/ShadowMail"
          target="_blank"
          rel="noopener noreferrer"
        >
          open source
        </a>.
        You can audit our code yourself to ensure you&apos;re comfortable with our practises.
      </span>
    `
  },
  {
    title: "Private",
    symbol: "fingerprint",
    description: `
      <span>
        We care about privacy as much as you do.  As such we never save, read or otherwise
        monitor your emails.
      </span>
    `
  }
];

import DescriptorElement from "@/components/DescriptorElement.vue";

export default {
  computed: {
    content: () => CONTENT
  },
  components: {
    DescriptorElement
  }
};
</script>
