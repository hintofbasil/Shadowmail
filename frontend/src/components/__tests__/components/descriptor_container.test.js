import React from 'react';
import renderer from 'react-test-renderer';

import DescriptorContainer from '../../descriptor_container';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <DescriptorContainer />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
