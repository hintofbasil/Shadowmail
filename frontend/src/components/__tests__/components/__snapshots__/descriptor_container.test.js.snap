// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
<div
  className="background"
>
  <div
    className="container"
  >
    <div
      className="row descriptor-container"
    >
      <div
        className="four columns descriptor descriptor-three"
      >
        <div
          className="descriptor-title"
        >
          Simple
        </div>
        <div>
          <i
            className="material-icons descriptor-icon"
          >
            done
          </i>
        </div>
        <div
          className="descriptor-text"
        >
          <span>
            ShadowMail is simple to use. Simply enter your email address and start using your private, virtual email address immediately.
          </span>
        </div>
      </div>
      <div
        className="four columns descriptor descriptor-three"
      >
        <div
          className="descriptor-title"
        >
          Secure
        </div>
        <div>
          <i
            className="material-icons descriptor-icon"
          >
            lock
          </i>
        </div>
        <div
          className="descriptor-text"
        >
          <span>
            ShadowMails is 100%
            <a
              href="https://github.com/hintofbasil/ShadowMail"
              rel="noopener noreferrer"
              target="_blank"
            >
              open source
            </a>
            . You can audit our code yourself to ensure you're comfortable with our practises.
          </span>
        </div>
      </div>
      <div
        className="four columns descriptor descriptor-three"
      >
        <div
          className="descriptor-title"
        >
          Private
        </div>
        <div>
          <i
            className="material-icons descriptor-icon"
          >
            fingerprint
          </i>
        </div>
        <div
          className="descriptor-text"
        >
          <span>
            We care about privacy as much as you do.  As such we never save, read or otherwise monitor your emails.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
